buildings/reference_data/other_reference.py

```diff
@@ -191,6 +195,105 @@ def update_imagery_survey_index(kx_api_key, dataset, dbconn):
     return "updated"
 
 
+def update_facilities(kx_api_key, dataset, dbconn):
+    if dataset != "nz_facilities":
+        return "error"
+
+    # get last update of layer date from log
+    from_var = last_update("facilities")
+
+    # current date
+    to_var = current_date()
+
+    cql_filter = ""
+    external_id = "facility_id"
+
+    layer = QgsVectorLayer(
+        URI.format(LDS_LAYER_IDS[dataset], kx_api_key, from_var, to_var, cql_filter)
+    )
+    if not layer.isValid():
+        # something went wrong
+        return "error"
+
+    if layer.featureCount() == 0:
+        return "current"
+
+    for feature in layer.getFeatures():
+        if feature.attribute("__change__") == "DELETE":
+            sql = "DELETE FROM buildings_reference.nz_facilities WHERE facility_id = %s;"
+            dbconn.execute_no_commit(sql, (feature[external_id],))
+
+        elif feature.attribute("__change__") == "INSERT":
```

Comment:
do we want to report to the user when it has tried to insert but the feature already exists? this just has logic so doesnt insetr if it already exists

Reply:
Theoretically that shouldn't happen.  The data is from the LDS which should be clean, and the table shouldn't get changed manually because it's only reference data.  Other reference data works the same way, so I wouldn't worry about adding that error check right now.